php/utils.php

```diff
@@ -544,11 +543,32 @@ function make_progress_bar( $message, $count, $interval = 100 ) {
 	return new \cli\progress\Bar( $message, $count, $interval );
 }
 
-function parse_url( $url ) {
-	$url_parts = \parse_url( $url );
+/**
+ * Helper function to use wp_parse_url when available or fall back to PHP's
+ * parse_url if not.
+ *
+ * Additionally, this adds 'http://' to the URL if no scheme was found.
+ *
+ * @param string $url       The URL to parse.
+ * @param int    $component The specific component to retrieve. Use one of the
+ *                          PHP predefined constants to specify which one.
+ *                          Defaults to -1 (= return all parts as an array).
+ * @return mixed False on parse failure; Array of URL components on success;
+ *               When a specific component has been requested: null if the
+ *               component doesn't exist in the given URL; a string or - in the
+ *               case of PHP_URL_PORT - integer when it does. See parse_url()'s
+ *               return values.
+ */
+function parse_url( $url, $component = - 1 ) {
+	if ( function_exists( 'wp_parse_url' ) ) {
+		$url_parts = wp_parse_url( $url, $component );
```

Comment:
IIRC, the `$component` parameter is only supported in WP since WP 4.7, so this may not work as expected.

Reply:
Added code to check for that.